qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateEnumAnnotationSupport.java

```diff
@@ -0,0 +1,116 @@
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * @TemplateEnum annotation support.
+ *
+ *
+ *               <code>
+ * &#64;TemplateEnum
+ * public enum Status {
+ * 		ON,
+ * 		OFF
+ * }
+ * </code>
+ *
+ * @see https://quarkus.io/guides/qute-reference#convenient-annotation-for-enums
+ *
+ */
+public class TemplateEnumAnnotationSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateEnumAnnotationSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+			TEMPLATE_ENUM_ANNOTATION
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+			SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (javaElement.getElementType() != IJavaElement.TYPE) {
+			return;
+		}
+		IType type = (IType) javaElement;
+		IAnnotation templateEnum = AnnotationUtils.getAnnotation((IAnnotatable) javaElement, TEMPLATE_ENUM_ANNOTATION);
+		if (templateEnum == null) {
+			return;
+		}
+
+		if (javaElement instanceof IType) {
+			IType type = (IType) javaElement;
+			collectResolversForTemplateEnum(type, templateEnum, context.getDataModelProject().getValueResolvers(),
+					monitor);
+		}
+	}
+
+	private static void collectResolversForTemplateEnum(IType type, IAnnotation templateExtension,
+			List<ValueResolverInfo> resolvers, IProgressMonitor monitor) {
+		try {
+			ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+			if (type.isEnum()) {
+				IField[] fields = type.getFields();
+				for (IField field : fields) {
+					IAnnotation fieldFromTemplateEnum = AnnotationUtils.getAnnotation(field, TEMPLATE_ENUM_ANNOTATION);
+					collectResolversForTemplateEnum(field,
+							fieldFromTemplateEnum != null ? fieldFromTemplateEnum : templateExtension, resolvers,
+							typeResolver);
+				}
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.SEVERE, "Error while getting methods of '" + type.getElementName() + "'.", e);
+		}
+	}
+
+	public static void collectResolversForTemplateEnum(IField field, IAnnotation templateExtension,
+			List<ValueResolverInfo> resolvers, IProgressMonitor monitor) {
+		ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(field);
+		collectResolversForTemplateEnum(field, templateExtension, resolvers, typeResolver);
+	}
+
+	private static void collectResolversForTemplateEnum(IField field, IAnnotation templateExtension,
+			List<ValueResolverInfo> resolvers, ITypeResolver typeResolver) {
+		ValueResolverInfo resolver = new ValueResolverInfo();
+		resolver.setSourceType(field.getDeclaringType().getFullyQualifiedName());
+		resolver.setSignature(typeResolver.resolveFieldSignature(field));
+		resolver.setNamespace(field.getDeclaringType().getFullyQualifiedName().replace('.', '_'));

```

Comment:
According the qute référence guide the namespace is just the simple  class name.